### DISCUSSION TITLE
Add support for the content-visibility CSS property to the image component

### DISCUSSION THREAD

**Discussion by atcastle** [vercel/next.js#64649] [Ideas]
**Describe the feature you'd like to request**

The Next.js image component should support the new content-visibility CSS property, which allows deferred rendering of offscreen images, potentially improving performance.

**Describe the solution you'd like**

I suggest we include the property in the rendered output of the image component. I'd leave the specifics of exactly which cases should use the property to the implementer.

**Describe alternatives you've considered**

There's no direct alternative for this property, as it does something that was not previously possible.

I've asked @joonpark13 to take a look at implementing this feature. I'll invite him to comment on this issue with more details.

Details on the content-visibility property: 
[web.dev](https://web.dev/content-visibility/)
[mdn](https://developer.mozilla.org/en-US/docs/Web/CSS/content-visibility)

**Comment by Joonpark13** [comment]
PR is up: #22141

**Comment by styfle** [comment]
I think with `next/future/image` we won't see much improvement for the default `placeholder="empty"` because that uses native lazy loading, however with `placeholder="blur"` I would expect to see perf improve 👍
